projects/igniteui-angular/src/lib/directives/mask/mask-parsing.service.ts

```diff
@@ -1,7 +1,18 @@
 import { Injectable } from '@angular/core';
 
-/** @hidden */
-export const MASK_FLAGS = ['C', '&', 'a', 'A', '?', 'L', '9', '0', '#'];
+
+const FLAGS = new Set('aACL09#&?');
+const REGEX = new Map([
+    ['C', /(?!^$)/u], // Non-empty
```

Comment:
Since this will match anything, pressing `space` will delete the prompt char and I don't think this is what we want.

Reply:
It is as per the [specification](https://github.com/IgniteUI/igniteui-angular/wiki/Mask-Directive-Specification#developer)